fe1-web/src/features/wallet/screens/WalletCreateSeed.tsx

```diff
@@ -1,57 +1,147 @@
 import STRINGS from 'resources/strings';
 
 import * as Wallet from '../objects';
+import { WalletStore } from '../store';
 
 const styles = StyleSheet.create({
-  smallPadding: {
-    padding: '1rem',
+  welcomeView: {
+    flex: 1,
+    backgroundColor: Colors.accent,
   } as ViewStyle,
 });
 
+type NavigationProps = StackScreenProps<
+  AppParamList,
+  typeof STRINGS.navigation_app_wallet_create_seed
+>;
+
 /**
  * Wallet screen to obtain a new mnemonic seed
  */
 const WalletCreateSeed = () => {
+  const navigation = useNavigation<NavigationProps['navigation']>();
+
+  const toast = useToast();
+
   /* used to set the mnemonic seed inserted by the user */
   const [seed, setSeed] = useState('');
 
-  // FIXME: Navigation should use a defined type here (instead of any)
-  const navigation = useNavigation<any>();
-
   useEffect(() => {
     setSeed(Wallet.generateMnemonicSeed());
   }, []);
 
+  useEffect(() => {
+    // Return the function to unsubscribe from the event so it gets removed on unmount
+    return navigation.addListener('focus', () => {
+      if (WalletStore.hasSeed()) {
+        navigation.navigate(STRINGS.navigation_app_home, {
+          screen: STRINGS.navigation_home_home,
+        });
+      }
+    });
+  }, [navigation]);
+
   const connectWithSeed = async () => {
     try {
       await Wallet.importMnemonic(seed);
-      navigation.reset({
-        index: 0,
-        routes: [{ name: STRINGS.navigation_wallet_home_tab }],
+      navigation.navigate(STRINGS.navigation_app_home, {
+        screen: STRINGS.navigation_home_home,
+      });
+    } catch (e) {
+      console.error(e);
+      toast.show(STRINGS.wallet_set_seed_error, {
+        type: 'danger',
+        placement: 'top',
+        duration: FOUR_SECONDS,
       });
-    } catch {
-      navigation.navigate(STRINGS.navigation_wallet_error);
     }
   };
 
   return (
-    <View style={containerStyles.centeredY}>
-      <TextBlock bold text={STRINGS.show_seed_info} />
-      <View style={styles.smallPadding} />
-      <CopiableTextInput text={seed} />
-      <View style={styles.smallPadding} />
-      <WideButtonView title={STRINGS.connect_with_this_seed} onPress={() => connectWithSeed()} />
-      <WideButtonView
-        title={STRINGS.back_to_wallet_setup}
-        onPress={() => navigation.navigate(STRINGS.navigation_wallet_setup_tab)}
-      />
+    <View style={styles.welcomeView}>
+      <ScreenWrapper>
+        <View style={containerStyles.centeredY}>
+          <Text style={[Typography.heading, Typography.negative]}>
+            {STRINGS.wallet_welcome_heading}
+          </Text>
+          <Text style={[Typography.paragraph, Typography.negative]}>
+            {STRINGS.wallet_welcome_text_first_time}
+          </Text>
+          <Text style={Typography.paragraph}>
+            <Text style={Typography.negative}>
+              {STRINGS.wallet_welcome_text_wallet_explanation_1}
+            </Text>
+            <Text> </Text>
```

Comment:
Are those empty `Text`s useful?

Reply:
Yes, they add spaces :)

Reply:
Isn't there a more straighforward solution? I'm thinking about `{'\n'}` like in [there](https://infinitbility.com/how-to-use-br-tag-in-react-native).

Reply:
This works as well but `prettier` auto formats it to this `<Text> </Text>` 😅

Reply:
I guess I could add a `<Space />` component?

Reply:
Oh ok I didn't know that. If `prettier` thinks that it's fine, I guess we can trust it :)